tests/lib/json/src/main.c

```diff
@@ -450,6 +486,130 @@ ZTEST(lib_json_test, test_json_obj_arr_decoding)
 	}
 }
 
+ZTEST(lib_json_test, test_json_2dim_arr_obj_encoding)
+{
+	struct obj_array_2dim obj_array_array_ts = {
+		.objects_array_array = {
+			[0] = {
+				.elements = {
+					[0] = { .name = "Simón Bolívar",   .height = 168 },
+					[1] = { .name = "Pelé",            .height = 173 },
```

Comment:
What's our policy on UTF-8 in source code?  IIRC that's a C11 feature, right?

Reply:
There is no policy on the usage of Unicode characters in source files, as far as I know (cc @tejlmand @nashif).

The support for non-ASCII characters in source files is implementation defined; but, I do not think there are any modern compilers (as in, released after 2000) that outright reject the source files that contain non-ASCII characters.

As for the UTF-8 encoded characters in the string literals, as far as I can see, they are never put through any locale-aware functions, and the validation inputs and outputs are in the same file (i.e. must use the same encoding), this should not be a problem in itself.

Reply:
It looks like C is slowly moving to accepting utf-8 in source code. As for source code, gcc has the `-finput-charset` and `-fexec-charset` command line options, which should probably be set to `UTF-8` to avoid problems building in locales using a different charset, in which case it's hard to know exactly what the compiler would do with utf-8 sequences in strings. I recall having adventures when building Java applications on Windows as the default charset there was UTF-16. Not sure what C compilers do in the Windows world.

Reply:
If you want to avoid all issues, you would need to use octal escaped sequences for utf-8 in strings. For tests, that's probably a reasonable plan.

Reply:
Updated the unit tests accordingly.